lib/liquid/tags/render.rb

```diff
@@ -38,13 +41,25 @@ def render_tag(context, output)
         parse_context: parse_context
       )
 
-      inner_context = context.new_isolated_subcontext
-      inner_context.template_name = template_name
-      inner_context.partial = true
-      @attributes.each do |key, value|
-        inner_context[key] = context.evaluate(value)
+      context_variable_name = @alias_name ? @alias_name : "this"
```

Comment:
To ease the adoption of `render` over `include` I think we decided to keep support for the template name fallback.